[quant][fx] Add node name as prefix to observer module name

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44846 [quant] Add quant APIs to save/load observer state_dict
* #44773 [quant][qat] Ensure fake_quant and observer can be disabled on scriptmodule
* **#44765 [quant][fx] Add node name as prefix to observer module name**

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23741355](https://our.internmc.facebook.com/intern/diff/D23741355)

supriyar committed:
[quant][fx] Add node name as prefix to observer module name

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -274,6 +275,7 @@ def is_observed(input_arg):
                 env[node.name] = observed_graph.node_copy(node, load_arg)
```
looks like this creates a new name, can you move L240 after this line and use:
```
self.activation_post_process_map[env[node.name]] = observer
```

jerryzh168 replied:
similar for the next case

jerryzh168 replied:
sorry, not this one

jerryzh168 replied:
the idea is the above, I'm not sure which line of code to change, please take a look.

supriyar replied:
@jerryzh168 why do we need to move L240 after this? This else clause doesn't insert observers.

supriyar replied:
The actual issue was somewhere else, in convert we explicitly check if module name startswith `activation_post_process`, so had to update that.

supriyar committed:
Update on "[quant][fx] Add node name as prefix to observer module name"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add node name as prefix to observer module name"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -463,7 +465,7 @@ def is_quantized(node):
             # handle activation post process calls
             if node.op == 'call_module':
-                if node.target.split('.')[-1].startswith('activation_post_process_'):
+                if '_activation_post_process_' in node.target.split('.')[-1]:
```
maybe put this in a function?
also i'm not sure whether it's a good idea to check the name actually after a second thought, can we check the type instead?

jerryzh168 reviewed (approved):
please move the checks to a function, optional: change it to check the type of the modules[node.target]

supriyar committed:
Update on "[quant][fx] Add node name as prefix to observer module name"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add node name as prefix to observer module name"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Add node name as prefix to observer module name"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_save_observer_state_dict

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23741355](https://our.internmc.facebook.com/intern/diff/D23741355)

[ghstack-poisoned]
